tools/cr/post_presubmit.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+# Copyright (c) 2024 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+"""
+This script posts comments to a GitHub PRs based on the results of a presubmit
+check.
+
+Example:
+    ./post_presubmit.py --input results.json --pr 1234
+
+
+You can produce the JSON output for a presubmit run by using:
+    npm run presubmit -- --base origin/master --json .presubmit_results.json
+"""
+
+import argparse
+from pathlib import Path
+import subprocess
+import json
+import sys
+import hashlib
+from typing import List, Dict, Any
+
+
+def post_comments(presubmit_entries: Dict[str, List[Dict[str, Any]]],
+                  pr_number: str) -> None:
+    """
+    Posts comments to a GitHub pull request based on presubmit results.
+
+    Args:
+        presubmit_entries:
+            A dictionary containing presubmit results categorized by severity
+            (e.g., 'errors', 'warnings', 'notifications').
+        pr_number:
+            The pull request number where the comments will be posted.
+    """
+    # Get a list of existing comments for this PR
+    existing_comments: List[str] = json.loads(
+        subprocess.check_output([
+            'gh', 'api', f'repos/brave/brave-core/issues/{pr_number}/comments',
```

Comment:
shouldn't it be `/pulls/` instead of `/issues/`?

I believe this will limit comments to 30 items, so you have to use pagination to get all comments.

Reply:
Fixed... apparently it is just a matter of passing `--paginate`. Thanks for catching that.

Reply:
> shouldn't it be /pulls/ instead of /issues/?

I was also confused about this, but no, a pull request is an issue in the github API for these purposes.